File: backend-lvlup/src/main/java/com/example/lvlupbackend/product/model/Product.java
package com.example.lvlupbackend.product.model;

import com.example.lvlupbackend.categories.model.Category;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "products")
@Data
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String name;
    private String description;
    private Integer price;
    private Integer stock;
    private String img;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
}

File: LVLUPfinal/app/src/main/java/com/example/lvlupfinal/data/products/Product.kt
package com.example.lvlupfinal.data.products

import com.example.lvlupfinal.data.categories.Category

data class Product(
    val id: Int?,
    val name: String,
    val description: String,
    val price: Int,
    val stock: Int,
    val img: String,          // ✅ ahora String
    val category: Category    // ✅ objeto Category
)